src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -75,75 +79,29 @@
-import static com.openlattice.chronicle.constants.EdmConstants.NOTIFICATION_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.OL_ID_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.PARTICIPANTS_PREFIX;
-import static com.openlattice.chronicle.constants.EdmConstants.PARTICIPATED_IN_AESN;
-import static com.openlattice.chronicle.constants.EdmConstants.PART_OF_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.PERSON_ID_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.PREPROCESSED_DATA_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.QUESTIONNAIRE_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.QUESTIONS_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.RECORDED_BY_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.RECORDED_DATE_TIME_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.RECORD_TYPE_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.RESPONDS_WITH_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.START_DATE_TIME_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.STATUS_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.STRING_ID_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.STUDY_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.TIMEZONE_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.TITLE_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.USED_BY_ENTITY_SET_NAME;
-import static com.openlattice.chronicle.constants.EdmConstants.USER_APPS_DICTIONARY;
-import static com.openlattice.chronicle.constants.EdmConstants.VALUES_FQN;
-import static com.openlattice.chronicle.constants.EdmConstants.VERSION_FQN;
-import static com.openlattice.chronicle.constants.OutputConstants.APP_PREFIX;
-import static com.openlattice.chronicle.constants.OutputConstants.DEFAULT_TIMEZONE;
-import static com.openlattice.chronicle.constants.OutputConstants.MINIMUM_DATE;
-import static com.openlattice.chronicle.constants.OutputConstants.USER_PREFIX;
+import static com.openlattice.chronicle.constants.EdmConstants.*;
+import static com.openlattice.chronicle.constants.OutputConstants.*;
 import static com.openlattice.edm.EdmConstants.ID_FQN;
 
 public class ChronicleServiceImpl implements ChronicleService {
     protected static final Logger logger = LoggerFactory.getLogger( ChronicleServiceImpl.class );
 
-    // studyId -> participantId -> deviceID -> device EKID
-    private final Map<UUID, Map<String, Map<String, UUID>>> studyDevices = new HashMap<>();
+    // appName -> orgId -> templateName -> entitySetID
+    private ImmutableMap<String, Map<UUID, Map<String, UUID>>> entitySetIdsByOrgId;
```

Comment:
Same comment as at the top -- this should be declared as a `Map`, not an `ImmutableMap`